src/app/clusters/electrical-energy-measurement-server/electrical-energy-measurement-server.cpp

```diff
@@ -126,6 +99,24 @@ namespace app {
 namespace Clusters {
 namespace ElectricalEnergyMeasurement {
 
+MeasurementData * MeasurementDataForEndpoint(EndpointId endpointId)
+{
+    auto index = emberAfGetClusterServerEndpointIndex(endpointId, app::Clusters::ElectricalEnergyMeasurement::Id,
+                                                      EMBER_AF_ELECTRICAL_ENERGY_MEASUREMENT_CLUSTER_SERVER_ENDPOINT_COUNT);
+
+    if (index == kEmberInvalidEndpointIndex)
+    {
+        return nullptr;
+    }
+
+    if (index >= EMBER_AF_ELECTRICAL_ENERGY_MEASUREMENT_CLUSTER_SERVER_ENDPOINT_COUNT)
+    {
+        ChipLogError(NotSpecified, "Internal error: invalid/unexpected energy measurement index.");
+        return nullptr;
+    }
```

Comment:
This length check is broken.  Why is this not comparing to the actuall array length of gMeasurements and instead comparing to some other constant?

Reply:
Was already there before this PR, but I'll add a fix into #31616